pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -188,13 +203,8 @@ func newLogicalReplicationWriterProcessor(
 		configByTable: procConfigByDestTableID,
 		spec:          spec,
 		getBatchSize: func() int {
-			// TODO(ssd): We set this to 1 since putting more than 1
-			// row in a KV batch using the new ConditionalPut-based
-			// conflict resolution would require more complex error
-			// handling and tracking that we haven't implemented
-			// yet.
 			if spec.Mode == jobspb.LogicalReplicationDetails_Immediate {
-				return 1
+				return int(immediateFlushBatchSize.Get(&flowCtx.Cfg.Settings.SV))
 			}
 			// We want to decide whether to use implicit txns or not based on
```

Comment:
We can maybe delete all this code about number of secondary indexes if the heuristic approach is able to just figure it out on the fly but I'll leave that to another day.